x-pack/plugins/actions/server/builtin_action_types/case/api.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { zipWith } from 'lodash';
+
+import {
+  ConnectorApi,
+  ExternalServiceParams,
+  ExternalServiceCommentResponse,
+  Comment,
+  PushToServiceResponse,
+  ConnectorApiHandlerArgs,
+} from './types';
+import { prepareFieldsForTransformation, transformFields, transformComments } from './utils';
+
+const handshakeHandler = async ({
+  externalService,
+  mapping,
+  params,
+}: ConnectorApiHandlerArgs) => {};
+const getIncidentHandler = async ({
+  externalService,
+  mapping,
+  params,
+}: ConnectorApiHandlerArgs) => {};
+
+const pushToServiceHandler = async ({
+  externalService,
+  mapping,
+  params,
+}: ConnectorApiHandlerArgs): Promise<PushToServiceResponse> => {
+  const { externalId, comments } = params;
+  const updateIncident = externalId ? true : false;
+  const defaultPipes = updateIncident ? ['informationUpdated'] : ['informationCreated'];
+  let currentIncident: ExternalServiceParams | undefined;
+  let res: PushToServiceResponse;
+
+  if (externalId) {
+    currentIncident = await externalService.getIncident(externalId);
+  }
+
+  const fields = prepareFieldsForTransformation({
+    params,
+    mapping,
+    defaultPipes,
+  });
+
+  const incident = transformFields({
+    params,
+    fields,
+    currentIncident,
+  });
+
+  if (updateIncident) {
+    res = await externalService.updateIncident({ incidentId: externalId, incident });
+  } else {
+    res = await externalService.createIncident({ incident });
+  }
+
+  if (
+    comments &&
+    Array.isArray(comments) &&
+    comments.length > 0 &&
+    mapping.get('comments')?.actionType !== 'nothing'
+  ) {
+    const commentsTransformed = transformComments(comments, ['informationAdded']);
+
+    // Create comments sequentially.
+    const promises = commentsTransformed.reduce(async (prevPromise, currentComment) => {
```

Comment:
I think this code will simplify some of our logic here, what do you think?

```
res.comments = [];
    for (const currentComment of commentsTransformed) {
      const comment = await externalService.createComment({
        incidentId: res.id,
        comment: currentComment,
        field: mapping.get('comments')?.target ?? 'comments',
      });
      res.comments = [
        ...(res.comments ?? []),
        {
          commentId: comment.commentId,
          pushedDate: comment.pushedDate,
        },
      ];
    }
```

Reply:
I like it! It's cleaner and easy to read. I tried to avoid for-loops but readiness always over complexity:)